packages/cursorless-engine/src/processTargets/modifiers/surroundingPair/index.ts

```diff
@@ -73,40 +71,6 @@ function processSurroundingPairCore(
   ] ?? [scopeType.delimiter];
 
   let node: SyntaxNode | null;
-  let textFragmentExtractor: TextFragmentExtractor;
-
-  const textFragmentScopeHandler =
-    languageDefinition?.getTextFragmentScopeHandler();
-
-  if (textFragmentScopeHandler != null) {
-    const containingScope = getContainingScopeTarget(
-      target,
-      textFragmentScopeHandler,
-      0,
-    );
-
-    if (containingScope != null) {
-      const surroundingRange = findSurroundingPairTextBased(
-        editor,
-        range,
-        containingScope[0].contentRange,
-        delimiters,
-        scopeType,
-      );
-      if (surroundingRange != null) {
-        // Found the pair within this text fragment or comment, e.g. "(abc)"
-        return surroundingRange;
-      }
-      // Search in the rest of the file, to find e.g. ("abc")
-      return findSurroundingPairTextBased(
```

Comment:
Fwiw the specific bug that this PR is trying to fix was on this line: we were directly falling back to the text-based algorithm when the input range was not in a text fragment, when instead it should have fallen back to parse-tree-based